File: src/api/league.ts
import apiClient from './request'
import type { LeagueStanding, TopScorer } from '@/types/league'

export default {
  /** 获取联赛积分榜 */
  getStandings(leagueId?: number, season?: string) {
    const params: Record<string, string | number> = {}
    if (leagueId) params.league_id = leagueId
    if (season) params.season = season
    return apiClient.get<LeagueStanding[]>('/league/standings', { params })
  },

  /** 获取射手榜 */
  getTopScorers(leagueId?: number, season?: string) {
    const params: Record<string, string | number> = {}
    if (leagueId) params.league_id = leagueId
    if (season) params.season = season
    return apiClient.get<TopScorer[]>('/league/top-scorers', { params })
  },
}

File: go_server/handlers/league.go
// Package handlers implements HTTP request handlers for league standings.
package handlers

import (
	"net/http"
	"strconv"

	"go_server/database"
	"go_server/models"

	"github.com/gin-gonic/gin"
)

// GetLeagueStandings returns standings for a given league.
// Query param ?leagueId=36 &season=2025-2026.
func GetLeagueStandings(c *gin.Context) {
	leagueIDStr := c.DefaultQuery("leagueId", "36")
	season := c.DefaultQuery("season", "2025-2026")

	leagueID, err := strconv.Atoi(leagueIDStr)
	if err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": "invalid leagueId"})
		return
	}

	var standings []models.LeagueStanding
	if err := database.DB.Where("league_id = ? AND season = ?", leagueID, season).
		Order("`rank` ASC").
		Find(&standings).Error; err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, standings)
}

// GetTopScorers returns the top scorers for a given league.
// Query param ?leagueId=36 &season=2025-2026 &type=goals|assists.
func GetTopScorers(c *gin.Context) {
	leagueIDStr := c.DefaultQuery("leagueId", "36")
	season := c.DefaultQuery("season", "2025-2026")

	leagueID, err := strconv.Atoi(leagueIDStr)
	if err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": "invalid leagueId"})
		return
	}

	orderField := "goals DESC"
	if c.Query("type") == "assists" {
		orderField = "assists DESC"
	}

	var scorers []models.TopScorer
	if err := database.DB.Where("league_id = ? AND season = ?", leagueID, season).
		Order(orderField).
		Limit(20).
		Find(&scorers).Error; err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, scorers)
}
